staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -106,7 +106,11 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	}
 
 	if redirect := resp.Header.Get("Location"); redirect != "" {
-		resp.Header.Set("Location", t.rewriteURL(redirect, req.URL, req.Host))
+		targetURL, err := url.Parse(redirect)
+		if err != nil {
+			return nil, errors.NewInternalError(fmt.Sprintf("error trying to parse Location header: %v", err))
```

Comment:
will this error include content of the invalid header? will echoing that be problematic?

Reply:
Yes, some errors may include it, but after looking into the URL package, it seems that all the errors containing the invalid header will have it quoted: https://github.com/golang/go/blob/master/src/net/url/url.go#L83-L93.

A practical example would be:
```go
_, err := url.Parse(`/redirected/target/%00<script>alert(1)</script>/%0`)
fmt.Println(err) // parse "/redirected/target/%00<script>alert(1)</script>/%0": invalid URL escape "%0"
```

So I don't think echoing it would be problematic.